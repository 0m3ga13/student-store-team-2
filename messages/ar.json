{
    "Index": {
        "Title": "المتجر الصغير",
        "Home": "الرئيسية",
        "Products": "المنتجات",
        "Orders": "الطلبيات",
        "State": "الحالة",
        "Buy": "شراء",
        "Borrow": "إعارة",
        "Description": "هذه المنصة هي مربحة للجميع: القرضاء يحققون دخلاً إضافيًا، والمقترضون يقللون التكاليف.",
        "Profile": "الحسـاب",
        "For Sale": "للبيع",
        "For Borrow": "للإعارة",
        "Sale & Borrow": "إعارة وشراء",
        "Day": "لليوم",
        "How It Works": "طريـقة العمل",
        "Hello": "مرحبا",
        "Sign out": "تسجيل الخروج",
        "Sign In": "تسجيل الدخول",
        "Login with Google": "سحل الدخول بحسابك",
        "Sign Up With Google": "سحل الدخول بحسابك"
    }
}
