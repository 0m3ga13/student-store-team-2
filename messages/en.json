{
    "Index": {
        "Title": "MiniStore",
        "Home": "Home",
        "Products": "Products",
        "Orders": "Orders",
        "State": "State",
        "Donate": "Donate",
        "Buy": "Buy",
        "Borrow": "Borrow",
        "Description": "This platform is a win-win for everyone involved lenders earn extra income, borrowers cut costs.",
        "Profile": "Profile",
        "for_sale": "for_sale",
        "for_borrow": "for_borrow",
        "Sale & Borrow": "Sale & Borrow",
        "Day": "Day",
        "How It Works": "How It Works",
        "recently added": "recently added",
        "Hello": "Hello",
        "Sign out": "Sign out",
        "Sign In": "Sign In",
        "Login with Google": "Login with Google",
        "Sign Up With Google": "Sign Up With Google",
        "Free delivery": "Free delivery",
        "feature descriprion 1": "You can receive your order without incurring any additional shipping costs",
        "qualite guarantee": "qualite guarantee",
        "feature descriprion 2": "Our products or services meet high standards and perform as promised.",
        "daily offers": "daily offers",
        "feature descriprion 3": "Exclusive discounts and promotions on a daily basis",
        "100% secure payement": "100% secure payement",
        "feature descriprion 4": "Your financial information is protected and your transactions are safe from unauthorized access",
        "My Dashboard": "My Dashboard",
        "Awaiting Pickup": "Awaiting Pickup",
        "All": "All",
        "Active": "Active",
        "Completed": "Completed",
        "Sort": "Sort:",
        "Newest": "Newest",
        "Oldest": "Oldest",
        "my stuff":"my stuff",
        "borrowings": "borrowings",
        "lendings": "lendings",
        "purchases": "purchases",
        "sales": "sales",
        "My Borrowings": "My Borrowings",
        "My Lendings": "My Lendings",
        "My Purchases": "My Purchases",
        "My Sales": "My Sales",
        "General details": "General details",
        "First Name": "First Name",
        "Last Name": "Last Name",
        "Birth Date": "Birth Date",
        "Gender": "Gender",
        "Institution": "Institution",
        "Contact & address Detail": "Contact & address Detail",
        "Phone Number": "Phone Number",
        "Building": "Building",
        "Street": "Street",
        "City": "City",
        "State_place": "State",
        "Save Changes": "Save Changes",
        "Upload Image": "Upload Image",
        "Close": "Close",
        "Upload": "Upload",
        "Loading": "Loading",
        "Not a member ?": "Not a member ?",
        "Sign Up": "Sign Up",
        "Store": "Store",
        "Create Your": "Create Your",
        "and start exchanging with other students": "and start exchanging with other students",
        "Or Login Using:": "Or Login Using:",
        "Login": "Login",
        "LOGIN": "LOGIN ",
        "Already a member ?": "Already a member ?",
        "Email and password are required": "Email and password are required",
        "Password should be at least 8 characters long": "Password should be at least 8 characters long",
        "and Conditions and Privacy Policy of The website": "and Conditions and Privacy Policy of The website",
        "Email": "Email",
        "Password": "Password",
        "Full Name": "Full Name",
        "Address": "Address",
        "and Privacy Policy": "and Privacy Policy",
        "I have read and agree with all of the Terms": "I have read and agree with all of the Terms",
        "PROFILE": "PROFILE",
        "SECURITY": "SECURITY",
        "Upload ID": "Upload ID",
        "Edit ID": "Edit ID",
        "Edit Image": "Edit Image",
        "WHY DONATE": "WHY DONATE",
        "MAKE A DIFFERENCE BY DONATING": "MAKE A DIFFERENCE BY DONATING",
        "Donate paragraph": "In our fast-paced lives, it's easy to forget the power of gratitude. Students, often on tight budgets, rely on our support. Your donation is more than just financial aid; it's a symbol of gratitude. By giving generously, you're helping students create a brighter future, one contribution at a time.",
        "Small Help": "Small Help",
        "Some Help": "Some Help",
        "Big Help": "Big Help",
        "List an Item": "List an Item",
        "Drag and drop some files here or click to select files": "Drag and drop some files here or click to select files",
        "Location": "Location",
        "Category": "Category",
        "books": "books",
        "kitchen": "kitchen",
        "sports": "sports",
        "games": "games",
        "clothes": "clothes",
        "electronics": "electronics",
        "furniture": "furniture",
        "description": "Description",
        "Product Condition": "Product Condition",
        "Sell": "Sell",
        "Price Per Week": "Price Per Week",
        "Price": "Price",
        "Upload Image(s)": "Upload Image(s)",
        "List": "List",
        "Product Name": "Product Name",
        "Comments": "Comments",
        "Type your comment": "Type your comment",
        "Type your reply": "Type your reply",
        "Respond": "Respond",
        "Like": "Like",
        "Send": "Send",
        "Authentication error": "Invalid login credentials. Please check your email and password.",
        "Filter": "Filter",
        "Pending": "Pending",
        "Quick Links": "Quick Links",
        "About LACITÉ": "ABOUT LACITÉ",
        "About Us": "ABOUT US",
        "Github Project Link": "GITHUB PROJECT LINK",
        "Contact Us": "Contact Us",
        "We Couldn't Find the page you're looking for": "We Couldn't Find the page you're looking for",
        "Go To Home": "Go To Home",
        "We Couldn't Find the requested Product": "We Couldn't Find the requested Product."
    }
}
