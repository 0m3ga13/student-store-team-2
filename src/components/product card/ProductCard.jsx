import Image from "next/image";
import Link from "next/link";
import { FaHandHoldingMedical } from "react-icons/fa";
import { PiShoppingCartThin } from "react-icons/pi";

export default function ProductCard() {
    // Placeholder object to be replaced with data from Firestore
    const products = [
        {
            id: 1,
            name: "Basic Tee",
            href: "#",
            imageSrc:
                "https://tailwindui.com/img/ecommerce-images/product-page-01-related-product-01.jpg",
            imageAlt: "Front of men's Basic Tee in black.",
            price: "$35",
            color: "Black",
            category: "Clothing",
            offer: ["borrow", "sale"],
        },
        // More products...
    ];

    // Conditional rendering function for displaying what icon to show under the item depending on its offer type
    const itemOffer = (product) => {
        return (
            <div>
                {product.offer.includes("sale") && (
                    <Link href='/dashboard'>
                        <p className='mt-1 flex justify-between text-sm font-small text-gray-900'>
                            <PiShoppingCartThin className=' text-xl text-blue-500' />{" "}
                            {product.price}
                        </p>
                    </Link>
                )}
                {product.offer.includes("borrow") && (
                    <Link href='/dashboard'>
                        <p className='mt-1 flex justify-between text-sm font-small text-gray-900'>
                            <FaHandHoldingMedical className=' text-lg text-blue-500' />{" "}
                            {product.price}
                        </p>
                    </Link>
                )}
            </div>
        );
    };

    return (
        <div className=' bg-white'>
            <div className='mx-auto max-w-2xl px-4 py-16 sm:px-6 sm:py-24 lg:max-w-7xl lg:px-8'>
                <div className='mt-6 grid grid-cols-1 gap-x-6 gap-y-10 sm:grid-cols-2 lg:grid-cols-4 xl:gap-x-8'>
                    {products.map((product) => (
                        <div key={product.id} className='group relative'>
                            <div className='relative w-full overflow-hidden rounded-md bg-gray-200  group-hover:opacity-75 lg:h-80'>
                                <Link href={`/product/${product.id}`}>
                                    <Image
                                        src={product.imageSrc}
                                        alt={product.imageAlt}
                                        width={500}
                                        height={500}
                                        className='h-full w-full object-cover object-center lg:h-full lg:w-full'
                                    />
                                </Link>
                                <p className='absolute bottom-0 left-0 mt-6 text-black'>
                                    6/10
                                </p>{" "}
                                {/*Replace with product state tag banner*/}
                            </div>
                            <div className='mt-3 flex justify-between'>
                                <div>
                                    <h3 className='text-sm text-gray-700'>
                                        <Link href='/category'>
                                            <div
                                                aria-hidden='true'
                                                className='absolute'
                                            />
                                            {product.category}
                                        </Link>
                                    </h3>
                                    <Link href={`/product/${product.id}`}>
                                        <p className='mt-1 text-lg font-medium text-gray-900'>
                                            {product.name}
                                        </p>
                                        {itemOffer(product)}
                                    </Link>
                                </div>
                                <Link href='/offer'>
                                    <div className='mt-1 text-sm font-medium text-gray-900'>
                                        product offer
                                    </div>{" "}
                                    {/*Replace with product offer tag button*/}
                                </Link>
                            </div>
                        </div>
                    ))}
                </div>
            </div>
        </div>
    );
}
