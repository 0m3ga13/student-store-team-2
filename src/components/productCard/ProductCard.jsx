"use client";

import Image from "next/image";
import Link from "next/link";
import { useTranslations } from "next-intl";
import { FaShoppingCart } from "react-icons/fa";
import { SiHandshake } from "react-icons/si";

import ItemConditionLabel from "./itemConditionLabel/ItemConditionLabel";
import ItemLabel from "./itemLabel/ItemLabel";

export default function ProductCard() {
    const t = useTranslations("Index");

    // Placeholder object to be replaced with data from Firestore
    const products = [
        {
            category: "Books",
            condition: "7",
            created_at: "October 16, 2023 at 3:33:59â€¯PM UTC+2",
            description: "lorem ipsum",
            for_borrow: true,
            for_sale: true,
            id: "0",
            imageSrc:
                "https://tailwindui.com/img/ecommerce-images/product-page-01-related-product-01.jpg",
            is_borrowed: true,
            is_sold: true,
            name: "Book",
            price: {
                borrow_price: "$20",
                sell_price: "$100",
            },
            uid: "121",
        },
    ];

    // Conditional rendering function for displaying what icon to show under the item depending on its offer type
    const itemOffer = (product) => {
        return (
            <div>
                {product.is_sold && (
                    <Link href='/dashboard'>
                        <div className='mt-1 font-jost flex justify-between text-base font-small text-titleContent'>
                            <FaShoppingCart className='text-xl mr-8 text-blue-500' />
                            {product.price.sell_price}
                        </div>
                    </Link>
                )}
                {product.is_borrowed && (
                    <Link href='/dashboard'>
                        <div className='mt-1 font-jost flex justify-between text-base font-small text-titleContent'>
                            <SiHandshake className='text-lg mr-8 text-blue-500' />{" "}
                            {product.price.borrow_price}/{t("Day")}
                        </div>
                    </Link>
                )}
            </div>
        );
    };
    // This determins what ItemLabel displays according the product offer type
    const determineLabel = (product) => {
        if (product.is_sold && product.is_borrowed) {
            return "Sale & Borrow"; // Translate the label
        } else if (product.is_sold) {
            return "For Sale"; // Translate the label
        } else if (product.is_borrowed) {
            return "For Borrow"; // Translate the label
        }
        return "";
    };

    return (
        <div className=' bg-white'>
            <div className='mx-auto max-w-2xl   lg:max-w-7xl '>
                {products.map((product) => (
                    <div key={product.id} className='group relative'>
                        <div className='relative w-full overflow-hidden rounded-md bg-gray-200  group-hover:opacity-75 lg:h-80'>
                            <Link href={`/product/${product.id}`}>
                                <Image
                                    src={product.imageSrc}
                                    alt={product.imageAlt}
                                    width={500}
                                    height={500}
                                    className='h-full w-full object-cover object-center lg:h-full lg:w-full'
                                />
                            </Link>
                            <div className='absolute bottom-0 left-0 mt-6 text-black'>
                                <ItemConditionLabel
                                    itemCondition={product.condition + "/10"}
                                />
                            </div>{" "}
                        </div>
                        <div className='mt-3 flex justify-between'>
                            <div>
                                <h3 className='font-jost text-sm font-semibold text-titleContent'>
                                    <Link href='/category'>
                                        <div aria-hidden='true' />
                                        {product.category}
                                    </Link>
                                </h3>
                                <Link href={`/product/${product.id}`}>
                                    <div className='mt-1 text-lg font-jost font-bold text-gray-900'>
                                        {product.name}
                                    </div>
                                    {itemOffer(product)}
                                </Link>
                            </div>
                            <Link href='/offerType'>
                                <ItemLabel
                                    textContent={determineLabel(product)}
                                />
                            </Link>
                        </div>
                    </div>
                ))}
            </div>
        </div>
    );
}
