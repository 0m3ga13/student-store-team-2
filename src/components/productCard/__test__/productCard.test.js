import renderer from "react-test-renderer";
import { NextIntlClientProvider } from "next-intl";
import messages from "../../../../messages/en.json";
import ProductCard from "../ProductCard";

it("renders correctly", () => {
    const tree = renderer
        .create(
            <NextIntlClientProvider locale='en' messages={messages}>
                <ProductCard />
            </NextIntlClientProvider>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
