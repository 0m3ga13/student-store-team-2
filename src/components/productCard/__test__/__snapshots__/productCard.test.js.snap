// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  className="group w-fit py-2 px-1 rounded-md pt-1 "
  href="/products/5"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onTouchStart={[Function]}
>
  <div
    className="relative rounded-md bg-gray-200  group-hover:opacity-75 overflow-hidden  w-[18rem] h-[21rem]"
  >
    <img
      alt="Classic literature set"
      className="object-cover object-center group-hover:scale-125 transition-all ease-in"
      data-nimg="fill"
      decoding="async"
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      sizes="100vw"
      src="/_next/image?url=https%3A%2F%2Fimages.placeholders.dev%2F%3Fwidth%3D100%26height%3D100&w=3840&q=75"
      srcSet="/_next/image?url=https%3A%2F%2Fimages.placeholders.dev%2F%3Fwidth%3D100%26height%3D100&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fimages.placeholders.dev%2F%3Fwidth%3D100%26height%3D100&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fimages.placeholders.dev%2F%3Fwidth%3D100%26height%3D100&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fimages.placeholders.dev%2F%3Fwidth%3D100%26height%3D100&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fimages.placeholders.dev%2F%3Fwidth%3D100%26height%3D100&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fimages.placeholders.dev%2F%3Fwidth%3D100%26height%3D100&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fimages.placeholders.dev%2F%3Fwidth%3D100%26height%3D100&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fimages.placeholders.dev%2F%3Fwidth%3D100%26height%3D100&w=3840&q=75 3840w"
      style={
        Object {
          "bottom": 0,
          "color": "transparent",
          "height": "100%",
          "left": 0,
          "objectFit": undefined,
          "objectPosition": undefined,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": "100%",
        }
      }
    />
    <div
      className="absolute bottom-0 left-0 mt-6 text-black"
    >
      <div
        className="relative"
      >
        <img
          alt="Item Condition"
          className="z-0 text-blue-400"
          data-nimg="1"
          decoding="async"
          height={39}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/conditionLabelVector.svg"
          style={
            Object {
              "color": "transparent",
            }
          }
          width={139}
        />
        <div
          className="font-jost font-extrabold text-sm text-bkg absolute bottom-1 left-2 w-fit"
        >
          $
        </div>
      </div>
    </div>
     
  </div>
  <div
    className="mt-3 flex "
  >
    <div
      className="product-infos"
    >
      <div
        className="mt-1 text-2xl font-jost  text-gray-900"
      >
        Classic literature set
      </div>
      <div
        className="font-jost text-sm  text-gray-600 mt-1"
      />
    </div>
    <div
      className="ml-auto  i"
    >
       
      <div
        className="capitalize flex justify-center text-center items-center font-jost font-bold rounded-full bg-secondaryGreen text-white text-sm w-[105px] h-[26px]"
      >
        Index
      </div>
    </div>
  </div>
</a>
`;
